packages/server/src/__tests__/plugin/usageReporting/stats.test.ts

```diff
@@ -472,3 +477,65 @@ describe('Check type stats', () => {
     expect(contextualizedStats).toMatchSnapshot();
   });
 });
+
+describe('Add trace to report', () => {
+  const defaultHeader = new ReportHeader({
+    hostname: 'hostname',
+    agentVersion: `@apollo/server`,
+    runtimeVersion: `node latest`,
+    uname: 'uname',
+    executableSchemaId: 'schema',
+    graphRef: 'graph',
+  });
+  const referencedFieldsByType = Object.create(null);
+  referencedFieldsByType['type'] = new ReferencedFieldsForType({
+    fieldNames: ['field1', 'field2'],
+    isInterface: false,
+  });
+
+  it('add as stats if asTrace is false', () => {
+    const report = new OurReport(defaultHeader);
+    report.addTrace({
+      statsReportKey: 'key',
+      trace: baseTrace,
+      asTrace: false,
+      referencedFieldsByType,
+    });
+
+    expect(report.tracesPerQuery['key']?.trace?.length).toBe(0);
+    expect(
+      Object.keys(report.tracesPerQuery['key']?.statsWithContext?.map).length,
+    ).toBe(1);
+  });
+
+  it('add as stats if asTrace is true but trace is too large', () => {
+    const report = new OurReport(defaultHeader);
+    (report as any).maxTraceBytes = 10;
```

Comment:
What if this were just an argument to `addTrace` with a default value?

Reply:
I'll update this but also we've just pushed a change to the reports handler that will mean we ignore any traces over 10mb on the ingest side (though they still get turned into stats), so I'll add a comment about that too.